guide_drafts/model-derives.md

```diff
@@ -0,0 +1,935 @@
+           `diesel_demo::models::Post : diesel::associations::HasTable`
+           `diesel_demo::models::Post : diesel::associations::BelongsTo<diesel_demo::models::User>`
+           `diesel_demo::models::Post : diesel::associations::BelongsTo<diesel_demo::models::User>`
+           `&diesel_demo::models::Post : diesel::BelongingToDsl<&[_]>`
+           `&diesel_demo::models::Post : diesel::associations::HasTable`
+           `&diesel_demo::models::Post : diesel::associations::BelongsTo<_>`
+           `&diesel_demo::models::Post : diesel::associations::BelongsTo<diesel_demo::models::User>`
+           `diesel_demo::models::Post : diesel::associations::HasTable`
+           `diesel_demo::models::Post : diesel::associations::HasTable`
+           `&diesel_demo::models::Post : diesel::associations::BelongsTo<diesel_demo::models::User>`
+           `&diesel_demo::models::Post : diesel::associations::BelongsTo<diesel_demo::models::User>`
+           `&diesel_demo::models::Post : diesel::associations::HasTable`
+           `&diesel_demo::models::Post : diesel::associations::BelongsTo<_>`
+           `&diesel_demo::models::Post : diesel::associations::BelongsTo<diesel_demo::models::User>`
+           `diesel_demo::models::Post : diesel::associations::HasTable`
+           `diesel_demo::models::Post : diesel::associations::HasTable`
+           `&diesel_demo::models::Post : diesel::associations::BelongsTo<diesel_demo::models::User>`
+           `&diesel_demo::models::Post : diesel::associations::BelongsTo<diesel_demo::models::User>`
+           `&mut diesel_demo::models::Post : diesel::BelongingToDsl<&[_]>`
+           `&mut diesel_demo::models::Post : diesel::associations::HasTable`
+           `&mut diesel_demo::models::Post : diesel::associations::BelongsTo<_>`
+           `&mut diesel_demo::models::Post : diesel::associations::BelongsTo<diesel_demo::models::User>`
+           `&mut diesel_demo::models::Post : diesel::associations::HasTable`
+           `&mut diesel_demo::models::Post : diesel::associations::BelongsTo<diesel_demo::models::User>`
+           `&mut diesel_demo::models::Post : diesel::associations::BelongsTo<diesel_demo::models::User>`
+           `&mut diesel_demo::models::Post : diesel::associations::HasTable`
+           `&mut diesel_demo::models::Post : diesel::associations::BelongsTo<_>`
+           `&mut diesel_demo::models::Post : diesel::associations::BelongsTo<diesel_demo::models::User>`
+           `&mut diesel_demo::models::Post : diesel::associations::HasTable`
+           `&mut diesel_demo::models::Post : diesel::associations::BelongsTo<diesel_demo::models::User>`
+           `&mut diesel_demo::models::Post : diesel::associations::BelongsTo<diesel_demo::models::User>`
+```
+
+In this particular case,
+we are using the `belongs_to()` method, which requires us to implement the `BelongsTo` trait.
+Below is what `cargo expand` generates for `#[derive(Associations)]`.
```

Comment:
Btw, is there a note about this mysterious `cargo expand` anywhere? I'm not sure many people know of it